src/Symfony/Component/Intl/Tests/LocaleTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Intl\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Intl\Locale;
+
+class LocaleTest extends TestCase
+{
+    public function provideGetFallbackTests()
+    {
+        $tests = array(
+            array('fr', 'fr_FR'),
+            array('fr', 'fr-FR'),
+            array('en', 'fr'),
+            array('root', 'en'),
+            array(null, 'root'),
+        );
+
+        if (function_exists('locale_parse')) {
```

Comment:
Shouldn't we also add these complex locales (`sl_Latn_IT`, etc.) to the normal tests when `locale_parse()` doesn't exist? The result would be of course different ... but I think we should test that too.

Reply:
I have added more tests.